sw/device/silicon_creator/rom_ext/rom_ext.c

```diff
@@ -621,33 +621,48 @@ static rom_error_t boot_svc_min_sec_ver_handler(boot_svc_msg_t *boot_svc_msg,
   const uint32_t current_min_sec_ver = boot_data->min_security_version_bl0;
   const uint32_t requested_min_sec_ver =
       boot_svc_msg->next_boot_bl0_slot_req.next_bl0_slot;
-  const uint32_t diff = requested_min_sec_ver - current_min_sec_ver;
 
   // Ensure the requested minimum security version isn't lower than the current
   // minimum security version.
   if (launder32(requested_min_sec_ver) > current_min_sec_ver) {
     HARDENED_CHECK_GT(requested_min_sec_ver, current_min_sec_ver);
-    HARDENED_CHECK_LT(diff, requested_min_sec_ver);
+    uint32_t max_sec_ver = current_min_sec_ver;
+
+    // Check the two flash slots for valid manifests and determine the maximum
+    // value of the new minimum_security_version.  This prevents a malicious
+    // MinBl0SecVer request from making the chip un-bootable.
+    const manifest_t *manifest = rom_ext_boot_policy_manifest_a_get();
+    rom_error_t error = rom_ext_verify(manifest, boot_data);
+    if (error == kErrorOk && manifest->security_version > max_sec_ver) {
+      max_sec_ver = manifest->security_version;
+    }
+    manifest = rom_ext_boot_policy_manifest_b_get();
+    error = rom_ext_verify(manifest, boot_data);
+    if (error == kErrorOk && manifest->security_version > max_sec_ver) {
+      max_sec_ver = manifest->security_version;
+    }
 
-    // Update boot data to the requested minimum BL0 security version.
-    boot_data->min_security_version_bl0 = requested_min_sec_ver;
+    if (requested_min_sec_ver <= max_sec_ver) {
```

Comment:
Something we can consider from a hardening perspective is whether it makes sense to add a crc over all the boot services commands. This is to avoid a fault being able to modify command parameters. 

We should review if this is something needed on a case by case basis.

Reply:
All Boot Service commands actually do have a "CRC" in the form of SHA256 digest: https://github.com/lowRISC/opentitan/blob/master/sw/device/silicon_creator/lib/boot_svc/boot_svc_header.h#L41